test/tests/outputs/checkpoint/tests

```diff
@@ -155,4 +155,25 @@
       detail = "be capable of restarting a simulation from the output data."
     []
   []
+  [autosave_timer]
+    type = RunApp
+    expect_out = "1 seconds have elapsed, autosaving checkpoint..."
+    input = autosavetimer.i
+    recover = false
+    max_parallel = 1
+    max_threads = 1
```

Comment:
Why the restrictions? Shouldn't this work perfectly fine with threads or MPI?

Reply:
Combining response with bottom comment -- the test file we are using is basically the same one used in the signal handler tests. Part of the reason I used those commands on those tests was to prevent the tests from running for a long time.

I can locally check how much time it takes to run these with parallel and threads and see.